packages/firestore/src/remote/remote_store.ts

```diff
@@ -177,7 +189,16 @@ export class RemoteStore implements TargetMetadataProvider {
       this.writeStream.stop();
 
       this.cleanUpWatchStreamState();
-      this.cleanUpWriteStreamState();
+
+      log.debug(
+        LOG_TAG,
+        'Stopping write stream with ' +
+          this.writePipeline.length +
+          ' pending writes'
+      );
+      // TODO(mikelehen): We only actually need to clear the write pipeline if
```

Comment:
I'm torn on this.
1. We're currently going out of our way to avoid filling writePipeline when the network is disabled (I could remove `this.writePipeline = []` here and the networkEnabled check in isWritePipelineFull() if we allowed writePipeline to persist when the network was disabled).
2. It means we do redundant work to re-load the same set of mutations from LocalStore when you re-enable the network.
3. I think it may be conceptually simpler to just consider writePipeline to be the "pre-fetched" mutations from the LocalStore, regardless of the network state... since they already persist across stream close / restarts.

So I'm inclined to leave it for now.